code/game/objects/items/weapons/twohanded.dm

```diff
@@ -605,6 +605,7 @@
 	wieldsound = 'sound/weapons/chainsawstart.ogg'
 	hitsound = null
 	armour_penetration = 35
+	block_chance = 80 //Melee only, only for 5 seconds after hittitng someone
```

Comment:
instead of doing this, would you flip the check in the `hit_reaction` to check `else if(owner.has_status....) => final_block_chance = 80)` this check, else is needed so it won't block projectiles.